### ISSUE TITLE
Is a duplicated ID between an element inside the `template` element and its use outside of `template` a real error?

### ISSUE THREAD

**Issue by juliemoynatPro** [validator/validator#1316]
Hi,

In a web page, I have a code inside a `template` element with an ID. This ID is then used once inside the page. When I copy/paste the generated code in the Validator, it's telling me there is an error: "Duplicate ID".

Code to reproduce error:

```html
<!DOCTYPE html>
<html lang="en">
<head>
  <title>Test</title>
</head>
<body>
  <p id="test">Test</p>
  
  <template>
    <p id="test"></p>
  </template>
</body>
</html>
```

Error and warning returned:

```
Error: Duplicate ID test.

[From line 11, column 3; to line 11, column 15](https://validator.w3.org/nu/#l11c15)

mplate>↩  <p id="test"></p>↩<

Warning: The first occurrence of ID test was here.

[From line 7, column 1; to line 7, column 13](https://validator.w3.org/nu/#l7c13)

d>↩<body>↩<p id="test">Test</
```

I'm not sure this is actually an error because, when I read the HTML specification, I've found this for [template syntax](https://html.spec.whatwg.org/multipage/syntax.html#template-syntax):

> The `template` element can have template contents, but such template contents are not children of the `template` element itself. Instead, they are stored in a `DocumentFragment` associated with a different `Document` — without a browsing context — so as to avoid the `template` contents interfering with the main `Document`.

Then for [template contents](https://html.spec.whatwg.org/multipage/scripting.html#template-contents), you have this:

>Each `template` element has an associated `DocumentFragment` object that is its template contents. The template contents have no conformance requirements.

And then for [no conformance requirements](https://html.spec.whatwg.org/multipage/dom.html#no-browsing-context), you have this:

>DOM nodes whose node document's browsing context is null are exempt from all document conformance requirements other than the HTML syntax requirements and XML syntax requirements.

So, is this really a duplicated ID?

**Comment by sideshowbarker** [comment]
@juliemoynatPro Thanks for having raised this, and sorry it took me so long to get around to fixing it

@zoltan-dulac Thanks for the bump — it otherwise would have taken me a while longer to get back through the list of open issues until I looked at this again

The fix for this is deployed now at https://validator.w3.org/nu/
